test/performance/latency/LatencyTestTypes.hpp

```diff
@@ -72,39 +81,30 @@ class LatencyType
 
 };
 
-inline bool operator ==(
-        const LatencyType& lt1,
-        const LatencyType& lt2)
-{
-    if (lt1.seqnum != lt2.seqnum)
-    {
-        return false;
-    }
-    if (lt1.data.size() != lt2.data.size())
-    {
-        return false;
-    }
-    for (size_t i = 0; i < lt1.data.size(); ++i)
-    {
-        if (lt1.data.at(i) != lt2.data.at(i))
-        {
-            return false;
-        }
-    }
-    return true;
-}
-
 class LatencyDataType : public eprosima::fastdds::dds::TopicDataType
 {
+    // Buffer size for size management
+    const uint32_t buffer_size_;
 public:
 
+
     LatencyDataType()
+        : buffer_size_(MAX_TYPE_SIZE - 4)
     {
         setName("LatencyType");
         m_typeSize = MAX_TYPE_SIZE;
         m_isGetKeyDefined = false;
     }
 
+    LatencyDataType(
+            const uint32_t& size)
+        : buffer_size_(size)
+    {
+        setName("LatencyType");
+        m_typeSize = size + 4;
```

Comment:
Use aligned size here:

```suggestion
        m_typeSize = 4 + ((size + 3) & ~3);
```

Reply:
Done in 759afcbf9cb2c567b30b3b3475617648754d5895

Does the default constructor need to be modified? I do not think it is being used with the refactor.

Reply:
No, it doesn't. It sets `m_typeSize = MAX_TYPE_SIZE` and `buffer_size_ = MAX_TYPE_SIZE - 4`, so alignment is ok.